pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go

```diff
@@ -52,33 +52,23 @@ func (p *DynamicPolicy) sharedCoresHintHandler(ctx context.Context,
 		return p.sharedCoresWithNUMABindingHintHandler(ctx, req)
 	}
 
-	ok, err := p.checkNormalShareCoresCpuResource(req)
-	if err != nil {
-		general.Errorf("failed to check share cores cpu resource for pod: %s/%s, container: %s",
-			req.PodNamespace, req.PodName, req.ContainerName)
-		return nil, fmt.Errorf("failed to check share cores cpu resource: %q", err)
-	}
-
-	if !ok {
-		_ = p.emitter.StoreInt64(util.MetricNameShareCoresNoEnoughResourceFailed, 1, metrics.MetricTypeNameCount, []metrics.MetricTag{
-			{
-				Key: "resource",
-				Val: "cpu",
-			},
-			{
-				Key: "userPodNs",
-				Val: req.PodNamespace,
-			},
-			{
-				Key: "userPodName",
-				Val: req.PodName,
-			},
-			{
-				Key: "userContainerName",
-				Val: req.ContainerName,
-			},
-		}...)
-		return nil, fmt.Errorf("no enough cpu resource")
+	if req.ContainerType == pluginapi.ContainerType_MAIN {
+		ok, err := p.checkNormalShareCoresCpuResource(req)
```

Comment:
would it be possible that vpa only scales sidecar?

Reply:
we will support sidecar follow main container to support this case in future